test/msw-testing-lib-react/src/App.test.tsx

```diff
@@ -0,0 +1,25 @@
+import { expect, it } from 'vitest'
+import { render, screen, userEvent, waitForElementToBeRemoved } from './utils/test-utils'
+import App from './App'
+
+it('Should return posts when clicking fetch button', async() => {
+  render(<App />)
+
+  expect(screen.getByRole('heading', { name: 'MSW Testing Library Example', level: 1 })).toBeDefined()
+
+  userEvent.click(screen.getByRole('button', { name: 'Fetch Posts' }))
+
+  await waitForElementToBeRemoved(() => screen.queryByLabelText('loading'))
+
+  //   First post
```

Comment:
Probably would be 'better' to import the JSON data, map through that and expect those `toBeDefined()`. That way if posts.json changes, you don't need to update the test

Reply:
Sounds good! :fire: